utils/core/src/main/kotlin/OrtAuthenticator.kt

```diff
@@ -129,19 +130,22 @@ internal fun getNetrcAuthentication(contents: String, machine: String): Password
 
     val iterator = lines.joinToString(" ").split(Regex("\\s+")).iterator()
 
-    var machineFound = false
+    var machineFound: String? = null
     var login: String? = null
     var password: String? = null
 
     while (iterator.hasNext()) {
-        when (iterator.next()) {
-            "machine" -> machineFound = iterator.hasNext() && iterator.next() == machine
-            "login" -> login = if (machineFound && iterator.hasNext()) iterator.next() else null
-            "password" -> password = if (machineFound && iterator.hasNext()) iterator.next() else null
-            "default" -> machineFound = true
+        when (val token = iterator.next()) {
+            "machine" -> machineFound = token.takeIf { iterator.nextOrNull() == machine }
+            "login" -> login = machineFound?.let { iterator.nextOrNull() }
+            "password" -> password = machineFound?.let { iterator.nextOrNull() }
+            "default" -> machineFound = token
         }
 
-        if (login != null && password != null) return PasswordAuthentication(login, password.toCharArray())
+        if (login != null && password != null) {
+            OrtAuthenticator.log.debug { "Found a '$machineFound' entry for '$machine'." }
+            return PasswordAuthentication(login, password.toCharArray())
+        }
```

Comment:
In order to allow a bit more detailed debugging, how instead logging in lines 140 / 141 like the following?

"Found login $login for machine $machine."
"Found a password for machine $machine."

Reply:
I didn't want to expose that much detail yet. Might be, though, that I'm forced to in a follow-up, if the current additions do not suffice for my current debugging.